### ISSUE TITLE
Update 'min(_:_:)' and 'max(_:_:)' gradients to match Python TensorFlow.

### ISSUE THREAD

**Issue by rxwei** [tensorflow/swift-apis#479]
IIRC, `tf.minimum` and `tf.maximum`'s gradients are not rounded.  We should update our implementation to match that.

**Comment by rxwei** [comment]
Note: Fixing this is a high priority.

**Comment by dan-zheng** [comment]
Here are the Python TensorFlow gradients for `minimum` and `maximum`, which we want to mimic:
```python
import tensorflow as tf
tf.enable_eager_execution()

x = tf.constant([4, 5, 3], dtype=tf.float32)
y = tf.constant([4, 2, 6], dtype=tf.float32)

with tf.GradientTape(persistent=True) as t:
  t.watch([x, y])
  result = tf.reduce_sum(tf.math.minimum(x, y))
print(t.gradient(result, [x, y]))
# tf.Tensor([1. 0. 1.], shape=(3,), dtype=float32)
# tf.Tensor([0. 1. 0.], shape=(3,), dtype=float32)

with tf.GradientTape(persistent=True) as t:
  t.watch([x, y])
  result = tf.reduce_sum(tf.math.minimum(y, x))
print(t.gradient(result, [x, y]))
# tf.Tensor([0. 0. 1.], shape=(3,), dtype=float32)
# tf.Tensor([1. 1. 0.], shape=(3,), dtype=float32)
```

Here is our current behavior, which is incorrect. I believe our logic was originally adapted from Tangent, which performs tie breaking (derivative is sometimes 0.5 before scaling).

```swift
import TensorFlow

let x = Tensor([4, 5, 3])
let y = Tensor([4, 2, 6])
print(gradient(at: x, y) { x, y in min(x, y).sum().scalarized() })
// ([0.5, 0.0, 1.0], [0.5, 1.0, 0.0])
print(gradient(at: x, y) { x, y in min(y, x).sum().scalarized() })
// ([0.5, 0.0, 1.0], [0.5, 1.0, 0.0])
```

**Comment by eaplatanios** [comment]
Thanks for clarifying @dan-zheng! In order to have all the relevant information here, the Python API gradient implementation can be found [here](https://github.com/tensorflow/tensorflow/blob/752556174a1dcf68693784a8c78f07ba818e230f/tensorflow/python/ops/math_grad.py#L1417).
